cibyl/config.py

```diff
@@ -43,27 +44,72 @@ def _ask_user_for_overwrite():
 
 
 class Config(UserDict):
-    """Representation of a Cybil's configuration file. Even though it starts
+    """Representation of a generic configuration file. Even though it starts
     without any contents, this dictionary can be filled in with the data
     from an external yaml file. No post-processing is performed on the read
     data, as this class acts as a direct interface between the system's file
     and the app.
     """
-
-    def load(self, file):
-        """Loads the contents of a file into this object. Any contents this
-        may beforehand have are lost and replaced by the data from the file.
-        In case of error, the contents are left untouched.
-
-        :raises YAMLError: If the file could not be parsed.
+    def load(self, path=None, data=None, file=None):
```

Comment:
I think this function could be simplified. Looking at its usage, only the 'path' argument is ever used on production code, 'data' and 'file' are only passed on tests. For that reason, I would remove those two arguments and just leave 'path'. Maybe call it 'load_from_path' or something alike. To be able to give it data for testing, make that part of the constructor. Have there the 'data' field which if not None will start with what we passed it. The 'field' argument I think we can drop altogether, unless I am missing somewhere it is being used.